app/models/pacbio/library.rb

```diff
@@ -86,5 +92,22 @@ def check_for_derived_aliquots
       errors.add(:base, 'Cannot delete a library that is used in a pool or run')
       throw(:abort)
     end
+
+    # Checks if the volume has been changed and if the new volume is greater than the used volume.
+    # If the volume has not been changed, the method immediately returns.
+    # If the new volume is greater than the used volume, the method returns true.
+    # If the new volume is less than or equal to the used volume
+    # - an error is added to the volume attribute and
+    # - the update operation is aborted.
+    #
+    # @return [TrueClass, nil]
+    # Returns true if the new volume is greater than the used volume, nil otherwise.
+    def check_volume
```

Comment:
would it be better to use the volume check method that you created in aliquotable?

Reply:
Please correct, if I am wrong.
I thought they are used for different purposes, where aliquotable volume check verifies whether the given volume is less than the available volume, and library volume check checks whether the given volume is less than the used volume.